### PR TITLE
Add documentation for https://github.com/eclipse/xtext-maven/issues/139

### PR DESCRIPTION
**Pull request by imhotep82**
https://github.com/eclipse/xtext-maven/issues/139

### CODE REVIEW

File: xtext-website/documentation/350_continuous_integration.md [hunk 1]
```diff
@@ -238,36 +237,85 @@ Now that we can build our language we need to be able to integrate our language
...
+
+```xml
+<plugin>
+	<groupId>org.eclipse.xtext</groupId>
+	<artifactId>xtext-maven-plugin</artifactId>
+	<version>${xtext-version}</version>
+	<executions>
+		<execution>
+			<goals>
+				<goal>generate</goal>
```
Comment (cdietrich) [inline xtext-website/documentation/350_continuous_integration.md:250]: dont we need more goals here?

Comment (imhotep82) [reply xtext-website/documentation/350_continuous_integration.md:250]: yes, you're right, thanks

Comment (cdietrich) [reply xtext-website/documentation/350_continuous_integration.md:250]: https://github.com/eclipse/xtext-maven/blob/b975d8174f2295dac3590b99cc34eb5708c43f76/org.eclipse.xtext.maven.plugin/src/test/resources/it/generate/pom.xml#L104

Comment (imhotep82) [reply xtext-website/documentation/350_continuous_integration.md]: I copied it from there.

Comment (imhotep82) [reply xtext-website/documentation/350_continuous_integration.md]: I have a strange diff, do you use tabs or spaces in your git config? Any special options I need to know?

Comment (cdietrich) [reply xtext-website/documentation/350_continuous_integration.md]: i have no idea. i would think "as is".
thus keep the spaces for the big table.
you may harmonize the maven snipptes to use spaces all over the place too.

File: xtext-website/documentation/350_continuous_integration.md [hunk 2]
```diff
@@ -238,6 +238,64 @@ Now that we can build our language we need to be able to integrate our language
...
+			<artifactId>my.javalanguage</artifactId>
+			<version>1.0.0-SNAPSHOT</version>
+		</dependency>
+	</dependencies>
+</plugin>
+```
+
+In this example, the option `installDslAsPrimarySource` is used to create traces that hide the underlying JAVA source and map the source file lines to the lines in the byte code.
+At least one source mapping is needed for the trace functionality to work as the plug-in has to map input to output folder locations.
+If you add the optional property `addOutputDirectoriesToCompileSourceRoots`, the output directories specified in `sourceMappings` will be appended as (test)compile roots of the current Maven project, which means the `maven-compiler-pluugin` will recognize them as JAVA source folders. Set to `false` to opt out, as `true` is the default.
```
Comment (cdietrich) [inline xtext-website/documentation/350_continuous_integration.md:297]: typo pluugin

File: xtext-website/documentation/350_continuous_integration.md [hunk 3]
```diff
@@ -238,6 +238,64 @@ Now that we can build our language we need to be able to integrate our language
...
+		<dependency>
+			<groupId>my.javalang</groupId>
+			<artifactId>my.javalanguage</artifactId>
+			<version>1.0.0-SNAPSHOT</version>
+		</dependency>
+	</dependencies>
+</plugin>
+```
+
+In this example, the option `installDslAsPrimarySource` is used to create traces that hide the underlying JAVA source and map the source file lines to the lines in the byte code.
```
Comment (cdietrich) [inline xtext-website/documentation/350_continuous_integration.md:295]: JAVA -> Java

File: xtext-website/_posts/releasenotes/2021-12-31-version-2-26-0.md [hunk 4]
```diff
@@ -21,6 +21,7 @@ TODO
 
 ## Enhancements
 
+- [xtext-maven#139](https://github.com/eclipse/xtext-maven/issues/139) : xtext-maven now automatically installs trace information for DSLs with a Java target.
```
Comment (cdietrich) [inline xtext-website/_posts/releasenotes/2021-12-31-version-2-26-0.md:24]: wonder if we should be more precise here about xbase usage

Comment (imhotep82) [reply xtext-website/_posts/releasenotes/2021-12-31-version-2-26-0.md:24]: Basically you could write a generator that produces java and not use xbase. But I could mention that as an example.

Comment (cdietrich) [reply xtext-website/_posts/releasenotes/2021-12-31-version-2-26-0.md:24]: yes and no. You wont have tracing information unless you make use of the TracingSugar / generateTracedFile feature.
(which 99% of the generators that produce java i have seen dont use)

Comment (imhotep82) [reply xtext-website/_posts/releasenotes/2021-12-31-version-2-26-0.md:24]: OK, i'll mention Xbase. People using custom java generators and manually using trace options should know what they're doing anyway.

**Comment by cdietrich** [comment]
LGTM @szarnekow what do you think?

File: xtext-website/documentation/350_continuous_integration.md [hunk 5]
```diff
@@ -238,6 +238,65 @@ Now that we can build our language we need to be able to integrate our language
...
+      <id>trace</id>
+      <goals>
+        <goal>install-debug-info</goal>
+        <goal>test-install-debug-info</goal>
+      </goals>
+    </execution>
+	</executions>
+	<configuration>
+    <addOutputDirectoriesToCompileSourceRoots>true</addOutputDirectoriesToCompileSourceRoots>
+		<languages>
```
Comment (szarnekow) [inline xtext-website/documentation/350_continuous_integration.md:267]: Indentation appears to use mixed tabs and spaces

**Review by szarnekow — APPROVED** [review]
Great! Thank you for taking the time and adding the documentation!

**Comment by cdietrich** [comment]
thx @imhotep82

**Pull request merged**